errors/src/main/java/com/palantir/conjure/java/api/errors/QoSExceptionReason.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.api.errors;
+
+import com.google.errorprone.annotations.CompileTimeConstant;
+
+final class QoSExceptionReason {
```

Comment:
Perhaps `QosReason`?
This needs to be public, otherwise nobody can use it :-)

Reply:
Updated the class to `public`. I've also renamed this to `QoSReason`, but I'm leaning slightly towards `QoSExceptionReason` since the class represents a reason for a quality of service exception rather than a reason for quality of service. I think either way, it's obvious what the use is for. No strong feelings.

Reply:
Ah, also noticed I've capitalized the s in `QoSReason`. Will lowercase for consistency.